tooling/lsp/src/requests/rename.rs

```diff
@@ -160,3 +160,144 @@ fn on_rename_inner(
+        let (mut state, noir_text_document) = setup().await;
+
+        let params = TextDocumentPositionParams {
+            text_document: lsp_types::TextDocumentIdentifier { uri: noir_text_document },
+            position: lsp_types::Position { line: 70, character: 3 }, // This is at the "t" of "test_multiple4"
+        };
+
+        let response = on_prepare_rename_request(&mut state, params)
+            .await
+            .expect("Could not execute on_prepare_rename_request");
+
+        assert_eq!(
+            response,
+            Some(PrepareRenameResponse::DefaultBehavior { default_behavior: true })
+        );
+    }
+
+    #[test]
+    async fn test_on_prepare_rename_request_can_be_applied_at_function_reference() {
+        let (mut state, noir_text_document) = setup().await;
+
+        let params = TextDocumentPositionParams {
+            text_document: lsp_types::TextDocumentIdentifier { uri: noir_text_document },
+            position: lsp_types::Position { line: 79, character: 4 }, // This is at the "t" of "test_multiple4"
+        };
+
+        let response = on_prepare_rename_request(&mut state, params)
+            .await
+            .expect("Could not execute on_prepare_rename_request");
+
+        assert_eq!(
+            response,
+            Some(PrepareRenameResponse::DefaultBehavior { default_behavior: true })
+        );
+    }
+
+    #[test]
+    async fn test_on_rename_request() {
+        let (mut state, noir_text_document) = setup().await;
+
+        let params = RenameParams {
+            text_document_position: TextDocumentPositionParams {
+                text_document: lsp_types::TextDocumentIdentifier { uri: noir_text_document },
+                position: lsp_types::Position { line: 79, character: 4 }, // This is at the "t" of "test_multiple4"
+            },
+            new_name: "renamed_test_multiple4".to_string(),
+            work_done_progress_params: WorkDoneProgressParams { work_done_token: None },
+        };
+
+        let response = on_rename_request(&mut state, params)
+            .await
+            .expect("Could not execute on_prepare_rename_request")
+            .unwrap();
+
+        let changes = response.changes.expect("Expected to find rename changes");
+        let mut changes: Vec<Range> = changes.values().flatten().map(|edit| edit.range).collect();
+        changes.sort_by_key(|range| range.start.line);
+        assert_eq!(
+            changes,
+            vec![
+                Range {
+                    start: Position { line: 70, character: 3 },
+                    end: Position { line: 70, character: 17 },
+                },
+                Range {
+                    start: Position { line: 79, character: 4 },
+                    end: Position { line: 79, character: 18 },
+                },
+                Range {
+                    start: Position { line: 94, character: 4 },
+                    end: Position { line: 94, character: 18 },
```

Comment:
It would be nice if we pulled these characters from the file and checked that these ranges contained the old name which we want to rename.

Reply:
Sounds good, I'll do it.

By the way, I got it to work with the elaborator 🎉 One part of the elaborator code seems to have been copied from the resolver, so I applied the same changes there and it worked 😄 

I think there's nothing to do on the `vscode-noir` side: once the rename thing is registered, "Rename symbol" shows up on the editor. I'll send a small video once I solve this particular issue.

Reply:
Oh, I saw some tests rely on the `7_functions` program... so I did the same thing. But it could be a bit brittle: if for some reason we change that example program, the lsp server might start to fail. But I don't know if existing test programs change at all. Do you think I should be using a separate program for these tests?